### PR TITLE
Add sleep precision tests [WIP]

### PR DESCRIPTION
**Pull request by LilithHafner**

### CODE REVIEW

File: test/misc.jl [hunk 1]
```diff
@@ -1239,3 +1239,67 @@ end
     @test (@allocations "a") == 0
     @test (@allocations "a" * "b") == 1
 end
+
+@testset "sleep precision" begin
+    println("GREP TARGET 1729 A")
```
Comment (Seelengrab) [inline test/misc.jl:1244]: I wonder if adding functionality for a logging blurb when a testset starts to run and ends to the basic testset directly would be good :thinking:

**Review by vtjnash — CHANGES_REQUESTED** [review]
Sleep is guaranteed only to return "eventually". It regularly takes many seconds longer than requested. This should not be tested for tolerances unless you have installed a custom RT kernel

**Comment by LilithHafner** [comment]
The primary reason I opened this PR is to get empirical results to inform discussion at #45144. It's also of note that windows sometimes sleeps for _less_ time than requested; I find that strange and disconcerting.

**Comment by vtjnash** [comment]
The Windows documentation for `Sleep` does make that explicit claim about returning early, but the documentation is wrong. I confirmed that with Microsoft developers here: https://github.com/MicrosoftDocs/sdk-api/blob/docs/sdk-api-src/content/synchapi/nf-synchapi-sleepex.md

**Pull request closed without merging**